src/Microsoft.Extensions.Logging/LoggerFactory.cs

```diff
@@ -41,11 +42,35 @@ public LoggerFactory(IConfiguration configuration)
                 throw new ArgumentNullException(nameof(configuration));
             }
 
+            UseConfiguration(configuration);
+        }
+
+        /// <summary>
+        /// Replaces the <see cref="IConfiguration"/> used for filtering.
```

Comment:
I'm a bit late on the design here, but why does this *replace* the configuration used for filtering? What if I have 2 sets of config that I want to use from 2 different sections? This seems to me like it should all be additive, always.